b2/console_tool.py

```diff
@@ -137,30 +118,64 @@ class Command(object):
     # Set to True for commands that receive sensitive information in arguments
     FORBID_LOGGING_ARGUMENTS = False
 
+    # The registry for the subcommands, should be reinitialized  in subclass
+    subcommands_registry = None
+
     def __init__(self, console_tool):
         self.console_tool = console_tool
         self.api = console_tool.api
         self.stdout = console_tool.stdout
         self.stderr = console_tool.stderr
 
     @classmethod
-    def add_subparser(cls, subparsers, parent):
-        parser = subparsers.add_parser(
-            mixed_case_to_hyphens(cls.__name__),
-            description=cls._format_description(cls.__doc__),
-            formatter_class=argparse.RawTextHelpFormatter,
-            parents=[parent]
-        )
-        cls._setup_subparser(parser)
-        return parser
+    def register_subcommand(cls, command_class):
+        assert cls.subcommands_registry is not None, 'Initialize the registry class'
+        return cls.subcommands_registry.register(key=mixed_case_to_hyphens(command_class.__name__)
+                                                )(command_class)
```

Comment:
please reformat it somehow to make it more readable, it's super weird now

Reply:
Fixed